uint8 c1_b
uint8 c2_b
uint8 c3_b

uint8 c1_t
uint8 c2_t
uint8 c3_t
